### DISCUSSION TITLE
Mastodon High Availability Setup?

### DISCUSSION THREAD

**Discussion by tkforty** [mastodon/mastodon#20158] [Q&A]
Has anybody done a further write-up on how to get do a high availability / load balancing setup? Ideally I'd like to have all of the servers components duplicated on at least 2 servers in different locations. I haven't found anything beyond the article below and I need to understand a little more on the Puma, Streaming API, and Sidekiq processes on how to cluster those between servers.

I see a lot of instance admin's really struggling with keeping the performance of their servers good and if someone who has a larger mastodon instance could go into detail on how to scale and when to scale it might help the overall performance and perception that people have of mastodon.

Mastodon is already getting a bit of a reputation for being "slow" and I think it's solely due to admin's not knowing when or how to scale. There are a lot of good instructions on how to setup a single server instance out there, but those people should know when to cutoff users or scale. For instance having some standards to know that a single server instance with a basic configuration and a specific server specifications can only handle 500 users (or something like that). . . 

https://docs.joinmastodon.org/admin/scaling/

**Comment by tuetenk0pp** [comment]
Theres some talk about a similar subject here #19500
